tracer/src/Datadog.Trace/OTelMetrics/MetricReader.cs

```diff
@@ -0,0 +1,138 @@
+
+#if NET6_0_OR_GREATER
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.OTelMetrics;
+
+internal static class MetricReader
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MetricReader));
+
+    private static System.Diagnostics.Metrics.MeterListener? _meterListenerInstance;
+    private static int _initialized;
+    private static int _stopped;
+
+    public static bool IsRunning =>
+        Interlocked.CompareExchange(ref _initialized, 1, 1) == 1 &&
+        Interlocked.CompareExchange(ref _stopped, 0, 0) == 0;
+
+    public static void Initialize()
+    {
+        if (Interlocked.CompareExchange(ref _initialized, 1, 0) == 1)
+        {
+            return;
+        }
+
+        var meterListener = new System.Diagnostics.Metrics.MeterListener();
+
+#if NET6_0 || NET7_0 || NET8_0
+        // Ensures instruments are fully de-registered on Dispose() for 6–8
+        // Static lambda => no captures/allocations
+        meterListener.MeasurementsCompleted = static (_, __) => { };
+#endif
+
+        meterListener.InstrumentPublished = MetricReaderHandler.OnInstrumentPublished;
+
+        meterListener.SetMeasurementEventCallback<byte>(MetricReaderHandler.OnMeasurementRecordedByte);
+        meterListener.SetMeasurementEventCallback<short>(MetricReaderHandler.OnMeasurementRecordedShort);
+        meterListener.SetMeasurementEventCallback<int>(MetricReaderHandler.OnMeasurementRecordedInt);
+        meterListener.SetMeasurementEventCallback<long>(MetricReaderHandler.OnMeasurementRecordedLong);
+        meterListener.SetMeasurementEventCallback<float>(MetricReaderHandler.OnMeasurementRecordedFloat);
+        meterListener.SetMeasurementEventCallback<double>(MetricReaderHandler.OnMeasurementRecordedDouble);
+
+        meterListener.Start();
+
+        Interlocked.Exchange(ref _meterListenerInstance, meterListener);
+        Interlocked.Exchange(ref _stopped, 0);
+
+        Log.Debug("MeterListener initialized successfully.");
+    }
+
+    public static void Stop()
+    {
+        var listener = Interlocked.Exchange(ref _meterListenerInstance, null);
+        if (listener is IDisposable disposableListener)
+        {
+            disposableListener.Dispose();
+            Interlocked.Exchange(ref _stopped, 1);
+            Interlocked.Exchange(ref _initialized, 0);
+            Log.Debug("MeterListener stopped.");
+        }
+    }
+
+    internal static void CollectObservableInstruments()
+    {
+        if (_meterListenerInstance != null)
+        {
+            try
+            {
+                _meterListenerInstance.RecordObservableInstruments();
+            }
+            catch (Exception ex)
+            {
+                Log.Warning(ex, "Error collecting observable instruments.");
+            }
+        }
+
+        private static async void ExportMetrics()
```

Comment:
HIGH-LEVEL: I think in general you will want to do everything synchronously instead of asynchronously, because ideally we would have our metrics reader / metrics exporter run synchronously on its own thread. That way we will always have a thread available for writing, even in the case that there's too many tasks trying to be assigned to threads.

Reply:
I think this part may take some additional insight and discussion from the rest of the team before converting to synchronous, so let's keep the async structure of your changes intact for now, no urgency to change this part of the design